classes/Cart.php

```diff
@@ -480,8 +480,10 @@ public function getCartRules($filter = CartRule::FILTER_ACTION_ALL, $autoAdd = t
                 WHERE `id_cart` = ' . (int) $this->id . '
                 ' . ($filter == CartRule::FILTER_ACTION_SHIPPING ? 'AND free_shipping = 1' : '') . '
                 ' . ($filter == CartRule::FILTER_ACTION_GIFT ? 'AND gift_product != 0' : '') . '
-                ' . ($filter == CartRule::FILTER_ACTION_REDUCTION ? 'AND (reduction_percent != 0 OR reduction_amount != 0)' : '')
-                . ' ORDER by cr.priority ASC, cr.gift_product DESC'
+                ' . ($filter == CartRule::FILTER_ACTION_REDUCTION ? 'AND (reduction_percent != 0 OR reduction_amount != 0)' : '') . '
+                AND `active` = 1
```

Comment:
Gift^^
                AND cr.`active` = 1
                AND cr.`quantity` > 0
                AND cr.`deleted` = 0
                AND cr.`date_from` <= "'.date('Y-m-d H:i:s').'"
                AND cr.`date_to` >= "'.date('Y-m-d H:i:s').'"

Reply:
Why is this code preferable?
```
AND cr.date_from <= "'.date('Y-m-d H:i:s').'"
AND cr.date_to >= "'.date('Y-m-d H:i:s').'"
```
In other places, `NOW() BETWEEN` is used anyway, e.g.:
https://github.com/PrestaShop/PrestaShop/blob/327c9e42b556fcd676ff02376c00d8d8c3521061/classes/CartRule.php#L1697-L1709

Reply:
1- https://www.sqlservercentral.com/blogs/is-between-faster-than-gte-and-lte
2- You forgot "active" and "deleted"
3- https://stackoverflow.com/questions/6113920/faster-to-use-mysqls-curdate-or-phps-date

To put it simply, a SQL comparison is faster if you give it the values ​​to compare directly rather than asking it to generate them.